### ISSUE TITLE
Warn about deprecated configurations like `compile`

### ISSUE THREAD

**Issue by bric3** [gradle/gradle#5217]
Have a build script that contains `compile` configuration

```groovy
// ...
dependencies {
    compile "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider"
    compile "org.springframework:spring-webmvc"
    compile "com.google.protobuf:protobuf-java"

    // ...

    testCompile "com.squareup.okhttp3:mockwebserver:$okhttpVersion"
```

### Expected Behavior

The console output warns about `compile` configuration, eventually linking to a guide explaining how to migrate (not just replacing `compile` by either `api` or `implementation`, but what to do with shared scripts, plugins, which one to migrate first if needed, etc...)

### Current Behavior

There's nothing.

### Context

While trying to add integration test, there's various recipes on the internet I don't know which one is the better or the future proof way. So I had to look at stackoverflow over here: https://stackoverflow.com/a/44493379/48136

### Your Environment
not needed

**Comment by eriwen** [comment]
@bric3 you nail the expected behavior here. I hope you find this new docs subsection (coming in Gradle 4.8) helpful right now: https://docs.gradle.org/nightly/userguide/building_java_projects.html#sec:java_dependency_management_overview — if not, please let us know how it could be more helpful.

@gradle/dependency-management if you intend to work on ways improve the UX as you are building so many new things, I hope you'll consider how to help users use the correct configurations in cases like this, and I would be happy for you to link to and improve [this doc](https://docs.gradle.org/nightly/userguide/building_java_projects.html#sec:java_dependency_management_overview) I linked above.

**Comment by jjohannes** [comment]
Addressed in: #9353
